lpc55_sign/src/signed_image.rs

```diff
@@ -242,3 +242,30 @@ pub fn generate_cfpa(
 
     Ok(cfpa)
 }
+
+pub fn remove_image_signature(mut img: Vec<u8>) -> Result<Vec<u8>, Error> {
+    let total_len = LittleEndian::read_u32(&img[0x20..0x24]);
+    let boot_field = LittleEndian::read_u32(&img[0x24..0x28]);
+    let cert_table_offset = LittleEndian::read_u32(&img[0x28..0x2c]);
+
+    if boot_field != BootImageType::SignedImage as u32 {
+        return Err(Error::NotSigned);
```

Comment:
I was going to ask for a way to detect if the image is signed, but this error variant is sufficient. Thanks!